File: ts-sdk/src/sdk/models/operations/getmetadatatemplatesid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetMetadataTemplatesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=template_id" })
  templateId: string;
}


export class GetMetadataTemplatesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetMetadataTemplatesIdPathParams;
}


export class GetMetadataTemplatesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  metadataTemplate?: shared.MetadataTemplate;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: sdks/go/pkg/models/operations/getmetadatatemplatesid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetMetadataTemplatesIDPathParams struct {
	TemplateID string `pathParam:"style=simple,explode=false,name=template_id"`
}

type GetMetadataTemplatesIDRequest struct {
	Retries    *utils.RetryConfig
	PathParams GetMetadataTemplatesIDPathParams
}

type GetMetadataTemplatesIDResponse struct {
	ClientError      *shared.ClientError
	ContentType      string
	MetadataTemplate *shared.MetadataTemplate
	StatusCode       int64
}
